api/user.auth.js

```diff
@@ -0,0 +1,56 @@
+import passport from 'passport';
+import jwt from 'jsonwebtoken';
+import { findUserByIdInDB } from '../service/users.service.js';
+import 'dotenv/config';
+
+const SECRET = process.env.SECRET;
+
```

Comment:
Trochę przekombinowane, zauważ że passport.authenticate zwraca ci już usera i możesz to zrobić tak
(piszę od ręki więc przetesuj to, ale raczej powinno działać)
```
const auth = async (req, res, next) => {
  passport.authenticate('jwt', { session: false }, (err, user) => {
    const token = req.get('Authorization').replace('Bearer ', '');
    
    if (!user || err || !token || token !== user.token) {
      return res.status(401).json({
        status: 'error',
        code: 401,
        message: 'Not authorized',
        data: 'Unauthorized',
      });
    }

    req.user = user;
    next();
  })(req, res, next);
};
```

Reply:
O! Super, tak zrobiłem - "z palca", czy nie - DZIAŁA! :) 
DZIĘKI! 👍